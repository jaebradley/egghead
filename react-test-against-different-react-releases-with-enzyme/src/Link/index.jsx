import React from 'react';

const Link = ({ address, hide }) => ((!hide && <a href={address}>Click</a>) || null);

export default Link;
