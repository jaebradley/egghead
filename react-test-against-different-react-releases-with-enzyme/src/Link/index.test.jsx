import React from 'react';
import {
  shallow,
  mount,
} from 'enzyme';
import toJSON from 'enzyme-to-json';

import Link from './';

describe('<Link /> shallow rendering', () => {
  const address = "www.google.com";

  it('a tag node renders href correctly', () => {
    const wrapper = shallow(<Link address={address} />);
    expect(wrapper.props().href).toBe(address);
  });

  it('returns null with true hide prop', () => {
    const wrapper = shallow(<Link hide={false} />);
    expect(wrapper.find('a').length).toBe(1);
    wrapper.setProps({ hide: true });
    expect(wrapper.get(0)).toBeNull();
  });

  it('matches the snapshot', () => {
    const wrapper = shallow(<Link />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});

describe('<Link /> mount rendering', () => {
  const address = "www.google.com";

  it('a tag node renders href correctly', () => {
    const wrapper = mount(<Link address={address} />);
    expect(wrapper.find('a').props().href).toBe(address);
    wrapper.unmount();
  });

  it('returns null with true hide prop', () => {
    const wrapper = mount(<Link hide={false} />);
    expect(wrapper.find('a').length).toBe(1);
    wrapper.setProps({ hide: true });
    expect(wrapper.find('a').length).toBe(0);
    wrapper.unmount();
  });
});
