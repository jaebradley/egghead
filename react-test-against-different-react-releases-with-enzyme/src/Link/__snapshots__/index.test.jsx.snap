// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> shallow rendering matches the snapshot 1`] = `
<a>
  Click
</a>
`;
